modules/csg/csg_shape.cpp

```diff
@@ -167,78 +169,244 @@ void CSGShape3D::_make_dirty(bool p_parent_removing) {
 	dirty = true;
 }
 
-CSGBrush *CSGShape3D::_get_brush() {
-	if (dirty) {
-		if (brush) {
-			memdelete(brush);
+enum ManifoldProperty {
+	MANIFOLD_PROPERTY_POSITION_X = 0,
+	MANIFOLD_PROPERTY_POSITION_Y,
+	MANIFOLD_PROPERTY_POSITION_Z,
+	MANIFOLD_PROPERTY_NORMAL_X,
+	MANIFOLD_PROPERTY_NORMAL_Y,
+	MANIFOLD_PROPERTY_NORMAL_Z,
+	MANIFOLD_PROPERTY_INVERT,
+	MANIFOLD_PROPERTY_SMOOTH_GROUP,
+	MANIFOLD_PROPERTY_UV_X_0,
+	MANIFOLD_PROPERTY_UV_Y_0,
+	MANIFOLD_MAX
+};
+
+static void _unpack_manifold(
```

Comment:
I still don't like bare structures, but had some difficulty with includes and symbols moving it to a class.

Suggestions?

Reply:
I like bare methods in the .cpp file, for things that don't need member access or being called from other files, this is quite nice. It's like the ultimate version of `private`.